app/src/addr.c

```diff
@@ -78,17 +83,25 @@ zxerr_t addr_getItem_internal(int8_t *displayIdx,
     }
 
     SCREEN(show_address_yes) {
+#if defined(TARGET_NANOS)
         snprintf(outKey, outKeyLen, "Verify if this");
         snprintf(outVal, outValLen, " public key was   added to");
+#else
+        snprintf(outKey, outKeyLen, "Warning:");
+        snprintf(outVal, outValLen, "Verify if this public key was added to");
+#endif
         return zxerr_ok;
     }
 
     SCREEN(show_address_yes) {
-        array_to_hexstr(outKey, outKeyLen, address_to_display.data, sizeof(address_to_display.data)); 
-        #if defined(TARGET_NANOX)
-            snprintf(outVal, outValLen, " using any Flow blockchain explorer.");
-        #else
+        #if defined(TARGET_NANOS)
+            array_to_hexstr(outKey, outKeyLen, address_to_display.data, sizeof(address_to_display.data)); 
             snprintf(outVal, outValLen, " using any Flow  blockch. explorer.");
+        #else
+            //Hack, but it does not seem dangerous and it saves a  bit of stack memory
```

Comment:
Are you referring to using "if defined()" as a hack? Is there a better way that would work here?

Reply:
No. This is about using outKey to calculate outVal, instead of using a local buffer variable. 

After giving it more thought, I changed the approach. Other screens require more stack, so this should not be a limitation. On the other hand, there is a risk that some future change could make the hex of address_to_display.data larger than outKeyLen, which would cause this code to fail. Thus now I think, this was a premature optimization.